charts/kube-prometheus-stack/templates/grafana/_helpers.tpl

```diff
@@ -0,0 +1,98 @@
+{{- define "kube-prometheus-stack.grafana.datasources" -}}
+{{- $scrapeInterval := .Values.grafana.sidecar.datasources.defaultDatasourceScrapeInterval | default .Values.prometheus.prometheusSpec.scrapeInterval | default "30s" -}}
+{{- $datasources := list -}}
+{{- if .Values.grafana.sidecar.datasources.defaultDatasourceEnabled }}
+{{/* Create jsonData dictionary first */}}
+{{- $jsonData := dict 
```

Comment:
interesting idea to use the dict + set function to build an structure, but not sure about maintainability.

Reply:
Yep indeed, when I look at it after 1 week, can't say for sure what it does , I would like to write tests for these functions but helm is pretty limited in this sphere(